src/main/groovy/6importCredentials.groovy

```diff
@@ -0,0 +1,232 @@
+Yaml yaml = new Yaml()
+
+globalDomain = Domain.global()
+credentialsStore =
+Jenkins.instance.getExtensionList(
+    'com.cloudbees.plugins.credentials.SystemCredentialsProvider')[0].getStore()
+
+String configPath = System.getenv("JENKINS_CONFIG_PATH")
+String configText = ''
+try {
+    configText = new File("${configPath}/credentials.yml").text
+} catch (FileNotFoundException e) {
+    logger.severe("Cannot find config file path @ ${configPath}/credentials.yml")
+    jenkins.doSafeExit(null)
+    System.exit(1)
+}
+
+// Delete any existing credentials to avoid clutter
+// and unintended duplication.
+credDomains = credentialsStore.getDomains()
+credDomains.each { domain ->
+    allCredentials = credentialsStore.getCredentials(domain)
+    allCredentials.each { credential ->
+        credentialsStore.removeCredentials(domain, credential)
+    }
+    // Now remove the domain
+    credentialsStore.removeDomain(domain)
+}
+
+List credentialConfig = yaml.load(configText)
+credentialConfig.each { newCredential ->
+    credentialType = newCredential.credentialType
+    id = newCredential.id
+    description = newCredential.description
+    scopeString = newCredential.scope
+
+    // All credentials must have a credentialType and scope
+    if (!credentialType || !scopeString) {
+        logger.severe("Missing data for credential. Please ensure every entry in the " +
+                      "credentials.yml file has a credentialType and scope")
+        jenkins.doSafeExit(null)
+        System.exit(1)
+    }
+
+    switch (scopeString.toLowerCase()) {
+        case 'global':
+            scope = CredentialsScope.GLOBAL
+            break
+        case 'system':
+            scope = CredentialsScope.SYSTEM
+            break
+        case 'user':
+            scope = CredentialsScope.USER
+            break
+        default:
+            logger.severe("Invalid scope specified for credential. Scope " +
+                          "should be GLOBAL, SYSTEM, or USER. Was: ${scopeString}")
+            jenkins.doSafeExit(null)
+            System.exit(1)
+    }
+
+    switch (credentialType) {
+        case 'usernamePassword':
+            username = newCredential.username
+            password = newCredential.password
+
+            // Make sure the username and password fields are not empty
+            if (!username || !password) {
+                logger.severe("Missing data for credential. Please ensure usernamePassword " +
+                              "entries in the credentials.yml file have a " +
+                              "username and password")
+                jenkins.doSafeExit(null)
+                System.exit(1)
+            }
+
+            cred = new UsernamePasswordCredentialsImpl(
+                scope,
+                id,
+                description,
+                username,
+                password
+            )
+            credentialsStore.addCredentials(
+                globalDomain,
```

Comment:
I believe Domain is more for just organizing the credentials. While the scope actually affects where they can be used.
```

System
    This credential is only available to the object on which the credential is associated. Typically you would use system-scoped credentials for things like email auth, slave connection, etc, i.e. where the Jenkins instance itself is using the credential. Unlike the global scope, this significantly restricts where the credential can be used, thereby providing a higher degree of confidentiality to the credential. 
Global
    This credential is available to the object on which the credential is associated and all objects that are children of that object. Typically you would use global-scoped credentials for things that are needed by jobs.
```

I defaulted Domain to Global as thats what we do in build jenkins. Perhaps I should add support to creating new domains? Also it looks like User isn't supported so I'm going to remove it